modules/Expectation.js

```diff
@@ -333,10 +333,21 @@ class Expectation {
     const expectedArgs = Array.prototype.slice.call(arguments, 0)
 
     assert(
+        spy.calls.length > 0,
+        ('spy was not called')
+    )
+
+
+    let actualArguments = ''
+    for(const arg of spy.calls) {
+      actualArguments =  actualArguments + '- ' + JSON.stringify(arg.arguments) + '\n'
+    }
+    const message = 'spy was never called with:\n%s.\n\nIt was called with:\n' + actualArguments
```

Comment:
One of the drawbacks to computing this message ahead of time like this is that we may not actually need it. For example, if the assertion passes, we will have done all this computation for nothing (i.e. we won't ever actually have to show this message). That's the great thing about the way our `assert` function works–it doesn't do any work at all if the assertion passes. It would be nice to keep this behavior so we don't slow tests down.

Also, please keep with existing code style everywhere (spacing, etc.)